akka-cluster/src/main/resources/reference.conf

```diff
@@ -280,6 +280,18 @@ akka {
       verbose-gossip-logging = off
     }
 
+    # Enforce configuration compatibility checks when joining a cluster.
+    # Set to off to allow joining nodes to join a cluster even when configuration incompatibilities are detected.
+    # This is particularly useful for rolling updates on clusters that do not support that feature. Since the old
+    # cluster won't be able to send the compatibility confirmation to the joining node, the joining node won't be able
+    # to 'know' if its allowed to join.
+    enforce-config-compat-on-join = on
+
+    # This section must contain full-qualified names JoinConfigCompatCheck
+    # each module can add its own checkers as long as the keys are unique.
+    configuration-compatibility-checkers {
+      # some-akka-module = "akka.cluster.SomeJoinConfigCompatChecker"
```

Comment:
that is a good structure

Reply:
I was about to write a checkers for `akka-cluster` but wasn't sure which properties to use. 

Maybe one possible candidate is the `downing-provider-class` and further SBR could also provide its own checker, for instance, that the same strategy is used everywhere.

WDYT?

Reply:
same downing-provider-class is a good candidate

same `state-store-mode` for cluster sharding is also good: https://github.com/akka/akka/blob/master/akka-cluster-sharding/src/main/resources/reference.conf#L74